virtcontainers/acrn_arch_base.go

```diff
@@ -0,0 +1,762 @@
+	appendConsole(devices []Device, path string) []Device
+
+	// appendImage appends an image to devices
+	appendImage(devices []Device, path string) ([]Device, error)
+
+	// appendBridges appends bridges to devices
+	appendBridges(devices []Device) []Device
+
+	// appendLPC appends LPC to devices
+	// UART device emulated by the ACRN-DM is connected to the system by the LPC bus
+	appendLPC(devices []Device) []Device
+
+	// appendSocket appends a socket to devices
+	appendSocket(devices []Device, socket types.Socket) []Device
+
+	// appendNetwork appends a endpoint device to devices
+	appendNetwork(devices []Device, endpoint Endpoint) []Device
+
+	// appendBlockDevice appends a block drive to devices
+	appendBlockDevice(devices []Device, drive config.BlockDrive) []Device
+
+	// handleImagePath handles the Hypervisor Config image path
+	handleImagePath(config HypervisorConfig)
+}
+
+type acrnArchBase struct {
+	path                 string
+	ctlpath              string
+	kernelParamsNonDebug []Param
+	kernelParamsDebug    []Param
+	kernelParams         []Param
+}
+
+const acrnPath = "/usr/bin/acrn-dm"
+const acrnctlPath = "/usr/bin/acrnctl"
+
+var baselogger *logrus.Entry
+
+// AcrnBlkDevPoolSz defines the number of dummy Blk
+// that will be created for hot-plugging virtio-blk
+// device.
+var AcrnBlkDevPoolSz = 8
+
+// AcrnBlkdDevSlot array provides translation between
+// the vitio-blk device index and slot it is currently
+// attached.
+// Allocating extra 1 to accommodate for VM rootfs
+// which is at driveIndex 0
+var AcrnBlkdDevSlot = make([]int, AcrnBlkDevPoolSz+1)
+
+// acrnKernelParamsNonDebug is a list of the default kernel
+// parameters that will be used in standard (non-debug) mode.
+var acrnKernelParamsNonDebug = []Param{
+	{"quiet", ""},
+}
+
+// acrnKernelParamsSystemdNonDebug is a list of the default systemd related
+// kernel parameters that will be used in standard (non-debug) mode.
+var acrnKernelParamsSystemdNonDebug = []Param{
+	{"systemd.show_status", "false"},
+}
+
+// acrnKernelParamsDebug is a list of the default kernel
+// parameters that will be used in debug mode (as much boot output as
+// possible).
+var acrnKernelParamsDebug = []Param{
+	{"debug", ""},
+}
+
+// acrnKernelParamsSystemdDebug is a list of the default systemd related kernel
+// parameters that will be used in debug mode (as much boot output as
+// possible).
+var acrnKernelParamsSystemdDebug = []Param{
+	{"systemd.show_status", "true"},
+	{"systemd.log_level", "debug"},
+	{"systemd.log_target", "kmsg"},
+	{"printk.devkmsg", "on"},
+}
+
+var acrnKernelRootParams = []Param{
+	{"root", "/dev/vda1 rw rootwait"},
+}
+
+var acrnKernelParams = []Param{
+	{"tsc", "reliable"},
+	{"no_timer_check", ""},
+	{"nohpet", ""},
+	{"console", "tty0"},
+	{"console", "ttyS0"},
+	{"console", "hvc0"},
+	{"log_buf_len", "16M"},
+	{"consoleblank", "0"},
+	{"iommu", "off"},
+	{"i915.avail_planes_per_pipe", "0x070F00"}, //this can also be passed via configuration.toml
```

Comment:
I'm not clear on the comment. Is this a note to yourself to add a new option to the config file?

Reply:
yes, currently I have not enabled the support for GVT-g. This is something that I am working in parallel. But have removed the comment.